app/src/main/java/com/bigyoshi/qrhunt/player/FragmentProfile.java

```diff
@@ -58,62 +50,43 @@
     private ArrayAdapter<PlayableQrCode> qrCodesAdapter;
     private FirebaseFirestore db;
 
-
-    /**
-     * Constructor method
-     *
-     * @param player          Current player
-     * @param lastDestination Previous navigation destination (by bottom navigation)
-     */
-    public FragmentProfile(Player player, int lastDestination){
-
-         this.playerInfo = player;
-         this.lastDestination = lastDestination;
-    }
-
-    /**
-     * Creates instance of fragment, and handles where the activity goes after pressing back button
-     * (eg, either to scanner or map)
-     *
-     * @param savedInstanceState SavedInstanceState
-     */
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
-
         super.onCreate(savedInstanceState);
-        getActivity().getOnBackPressedDispatcher().addCallback(this,
-                new OnBackPressedCallback(true) {
-
-            @Override
-            public void handleOnBackPressed() {
-
-                Intent intent = new Intent(getContext(), MainActivity.class);
-                Bundle prevNav = new Bundle();
-                prevNav.putSerializable("previous", lastDestination);
-                intent.putExtras(prevNav);
-                startActivity(intent);
-            }
-        });
+        getActivity()
+                .getOnBackPressedDispatcher()
+                .addCallback(
+                        this,
+                        new OnBackPressedCallback(true) {
+
+                            @Override
+                            public void handleOnBackPressed() {
+
+                                Intent intent = new Intent(getContext(), MainActivity.class);
+                                Bundle prevNav = new Bundle();
+                                prevNav.putSerializable("previous", lastDestination);
+                                intent.putExtras(prevNav);
+                                startActivity(intent);
+                            }
+                        });
```

Comment:
honestly hate this style of formatting, i wish there was good java auto formatters out there

but yeah this was my own doing

Reply:
Oh, by the way. I am trying to customize how the error message looks, so it doesn't look so out of place.

Reply:
Oh wow this is very annoying. It's a stretch goal now.